misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/LineButton.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { push } from 'connected-react-router';
+
+const mapStateToProps = ({ appHandler }) => ({
+  appId: appHandler.activeApplication ? appHandler.activeApplication.id : null,
+});
+
+function LineButton(WrappedComponent) {
+  class Wrapped extends PureComponent {
+    handleClick = () => {
+      const { wfProcessId, activityId, lineId } = this.props;
+      const { dispatch } = this.props;
+
+      const location = `/workflow/${wfProcessId}/activityId/${activityId}/lineId/${lineId}`;
+      dispatch(push(location));
```

Comment:
map 'push' into mapDispatchToProps and call it from props

Reply:
Can't do it as LineButton is not connected to the store, and parent component can't use map-assignment due to the fact that child components can have different actions.